cmd/server/main.go

```diff
@@ -1,31 +1,93 @@
 package main
 
 import (
+	"fmt"
 	"github.com/DimaKoz/practicummetrics/internal/common/config"
+	"github.com/DimaKoz/practicummetrics/internal/common/repository"
 	"github.com/DimaKoz/practicummetrics/internal/server/handler"
+	middleware2 "github.com/DimaKoz/practicummetrics/internal/server/middleware"
 	"github.com/labstack/echo/v4"
-	"log"
+	"github.com/labstack/echo/v4/middleware"
+	"go.uber.org/zap"
+	"os"
+	"time"
 )
 
+var sugar zap.SugaredLogger
+
 func main() {
 
+	logger, err := zap.NewDevelopment()
+	if err != nil {
+		panic(err)
+	}
+	defer func(logger *zap.Logger) {
+		err := logger.Sync()
+		if err != nil {
+			panic(err)
+		}
+	}(logger)
+
+	sugar = *logger.Sugar()
+
 	cfg, err := config.LoadServerConfig()
 	if err != nil {
-		log.Fatalf("couldn't create a config %s", err)
+		sugar.Fatalf("couldn't create a config %s", err)
 	}
 
 	// from cfg:
-	log.Println("cfg:")
-	log.Println("address:", cfg.Address)
+	sugar.Info(
+		"cfg: \n", cfg.String(),
+	)
+	sugar.Infow(
+		"Starting server",
+	)
+	_, err = os.Stat(cfg.FileStoragePath)
+	if os.IsNotExist(err) {
+		cfg.Restore = false
+		fmt.Printf("%v file does not exist\n", cfg.FileStoragePath)
+	}
+	repository.SetupFilePathStorage(cfg.FileStoragePath)
+	if cfg.Restore && cfg.FileStoragePath != "" {
+		if err = repository.Load(); err != nil {
+			sugar.Fatalf("couldn't restore metrics by %s", err)
+		}
+	}
+
+	if cfg.FileStoragePath != "" {
+		if cfg.StoreInterval != 0 {
+			handler.SyncSaveUpdateHandlerJSON = false
+			ticker := time.NewTicker(time.Duration(cfg.StoreInterval) * time.Second)
+			defer ticker.Stop()
+			go func() {
+				for range ticker.C {
+					err = repository.Save()
+					if err != nil {
+						sugar.Fatalf("server: cannot save metrics: %s", err)
+					}
+				}
+			}()
+		} else {
+			handler.SyncSaveUpdateHandlerJSON = true
+		}
+	}
 
 	e := echo.New()
+	e.Use(middleware.RequestLoggerWithConfig(middleware2.GetRequestLoggerConfig(sugar)))
+	e.Use(middleware.BodyDump(func(c echo.Context, reqBody, resBody []byte) {
+		sugar.Infow(
+			"body:", "reqBody:", string(reqBody[:]),
+		)
+	}))
+	e.Use(middleware2.GetGzipMiddlewareConfig())
+
 	e.POST("/update/:type/:name/:value", handler.UpdateHandler)
```

Comment:
It would be better, if you move to another package, for example, `internal/app/handle.go`

Reply:
```
func Router() http.Handler {
	r := chi.NewRouter()

	r.Post("/", shortener)
	r.Get("/{id}", expander)

	return r
}
```

Reply:
done